src/test/java/io/kestra/plugin/serdes/avro/infer/InferAvroSchemaFromIonTest.java

```diff
@@ -0,0 +1,176 @@
+                               "fields": [
+                                 {
+                                   "name": "myName",
+                                   "type": [
+                                     "string",
+                                     "null"
+                                   ]
+                                 }
+                               ]
+                             },
+                             "null"
+                           ]
+                         },
+                         "null"
+                       ]
+                     }
+                   ]
+                 }
+                """
+        );
+    }
+
+    @Test
+    void array_of_objects_with_unmatching_types() throws IOException {
+        this.run("""
+                {
+                    myArray: [
+                        {myName: "one"},
+                        {myName: "two", anAdditionalField: "hey"},
+                        {myName: "three"}
+                    ]
+                }
+                """,
+            """
+                {
+                  "fields": [
+                      {
+                        "name": "myArray",
+                        "type": [
+                          {
+                            "type": "array",
+                            "items": [
+                              "null",
+                              {
+                                "type": "record",
+                                "fields": [
+                                  {
+                                    "name": "myName",
+                                    "type": [
+                                      "null",
+                                      "string"
+                                    ]
+                                  },
+                                  {
+                                    "name": "anAdditionalField",
+                                    "type": [
+                                      "string",
+                                      "null"
+                                    ]
+                                  }
+                                ]
+                              }
+                            ]
+                          },
+                          "null"
+                        ]
+                      }
+                    ]
+                }
+                """
+        );
+    }
+
+    void run(String input, String expected) throws IOException {
+        var output = new ByteArrayOutputStream();
+
+        // when
+        new InferAvroSchema().inferAvroSchemaFromIon(
+            new InputStreamReader(new ByteArrayInputStream(input.getBytes())),
+            output
+        );
+
+        // then
+        try (ByteArrayInputStream in = new ByteArrayInputStream(output.toByteArray())) {
+            var resultingAvroSchema = new String(in.readAllBytes());
+            JSONAssert.assertEquals(
```

Comment:
Maybe you can just compare the fingerprint of the schema: `SchemaNormalization.parsingFingerprint64`

Reply:
will try to see what it looks like, my goal was to make it easier to debug